ObjectiveGit/GTTreeEntry.m

```diff
@@ -39,6 +39,7 @@
 
 @interface GTTreeEntry ()
 @property (nonatomic, assign, readonly) const git_tree_entry *git_tree_entry;
+@property (nonatomic, assign, readonly) git_tree_entry* entry_to_free;
```

Comment:
After a closer look, [git_tree_entry_byname](https://libgit2.github.com/libgit2/#HEAD/group/tree/git_tree_entry_byname) returns a pointer that is managed internally. As such, there is no need to call `git_tree_entry_free` on it, so the code in `-[GTTree treeEntryByPath:]` above can be changed to the simpler version `-entryWithName:` uses.